src/main/java/stepdefinitions/DefinitionSteps.java

```diff
@@ -97,12 +104,43 @@ public void anonymousUserSetsDataWithFollowingValues(final DataTable dataTable)
                     System.out.println("field not found");
             }
         }
-//        signInPage.enterTextToFieldEmail(dataTable.column(0).get(0));
-//        signInPage.enterTextToFieldPassword(dataTable.column(1).get(0));
     }
 
     @Then("User click by the button Sign In")
     public void clickByButtonSignIn() {
         signInPage.clickButtonSignIn();
     }
+
+    @And("User verify the visibility list of the {int} products")
+    public void verifyVisibilityListOfProducts(final int productsCount) {
+        List<WebElement> products = productPage.getProductsCards();
+        assertEquals(products.size(), productsCount);
```

Comment:
Add an error message to your Assertions, for example:

Assert.assertEquals(actualResult, expectedResult, String.format("Expected number of products '%s' but actual is '%s'",actualResult, expectedResult));